tests/abi_roundtrip.py

```diff
@@ -0,0 +1,145 @@
+from typing import Generic, TypeVar, cast
+
+import pyteal as pt
+from pyteal import abi
+
+from tests.blackbox import Blackbox, BlackboxWrapper, PyTealDryRunExecutor
+
+T = TypeVar("T", bound=abi.BaseType)
+
+DEFAULT_DYNAMIC_ARRAY_LENGTH = 3
+
+
+class ABIRoundtrip(Generic[T]):
+    def __init__(
+        self,
+        t: type[T] | None,
+        length: int | None = None,
+        annotation_instance: abi.BaseType = None,
+    ):
+        self.annotation: type[abi.BaseType]
+        self.type_spec: abi.TypeSpec
+        self.instance: abi.BaseType
+
+        if annotation_instance:
```

Comment:
@tzaffi While trying to understand the branching here, I experimented with local refactorings that I've opened as an optional child PR:  https://github.com/algorand/pyteal/pull/349.

Can you have a look to determine if it's worth pulling in or closing?

Separately and unrelated, I batched a couple of minor, optional refactorings into https://github.com/algorand/pyteal/pull/346.  I feel less strongly about #346 though want to add it to your radar for review.

Reply:
#346 merged 🙏

Reply:
#346 merged as well. Great catch and improvement!!!!